src/SDK/Common/Configuration/Configuration.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Configuration;
+
+/**
+ * Configuration can come from one or more of the following sources (from highest to lowest priority):
+ * - values defined in php.ini
+ * - environment variable
+ * - configuration file (todo)
+ */
+class Configuration
+{
+    private static ?array $resolvers = null;
+
+    /**
+     * @return array<Resolver>
+     */
+    private static function resolvers(): array
+    {
+        self::$resolvers ??= [
+            new IniResolver(),
+            new EnvironmentResolver(),
+        ];
+
+        return self::$resolvers;
+    }
+
+    public static function has(string $name): bool
+    {
+        foreach (self::resolvers() as $resolver) {
+            if ($resolver->hasVariable($name)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static function getResolver(string $name): Resolver
+    {
+        foreach (self::resolvers() as $resolver) {
+            if ($resolver->hasVariable($name)) {
+                return $resolver;
+            }
+        }
+
+        return DefaultResolver::instance();
+    }
+
+    public static function getInt(string $key, int $default): int
```

Comment:
Do we need the `$default` parameters or could we return `null` instead?

Reply:
We could return `null`, it would mean changing most of our factories to match. Providing a default has the benefit that the default is validated against known types and possible values, so seems safer to me.

Reply:
I missed that we throw if the key is not set and no default is given - changing the default parameter to nullable / default null to match the other methods is sufficient then.

Reply:
I'm not sure about OTel specifically, but in the past it's been important on occasion to be able to distinguish between "not set" and "default", when there are multiple things that control something. Usually this comes up because of legacy env vars. For instance, and this is totally made up for example: `OTEL_HOST=collector.local` but then later we realize that means it can't support `https` as is, so there's a new one `OTEL_COLLECTOR_URI=https://collector.local`. For the sake of falling back to an older config, you generally want to be able to distinguish between default and unset.

Reply:
@morrisonlevi we currently manage that scenario with `::has`, although it might start to get gnarly with more variables:
```php
$endpoint = Configuration::has(Variables::OTEL_EXPORTER_OTLP_TRACES_ENDPOINT)
    ? Configuration::getString(Variables::OTEL_EXPORTER_OTLP_TRACES_ENDPOINT)
    : Configuration::getString(Variables::OTEL_EXPORTER_OTLP_ENDPOINT);
```